docs/ADR-4-Era-selectors-for-CLI-commands.md

```diff
@@ -0,0 +1,97 @@
+
+Going foward we have additional concerns to consider:
+
+Some commands require access to the protocol parameters in the era the command is operating in and the contents of the protocol parameters could have meaningful bearing on how any transactions derived from those commands will be interpreted on-chain.
+
+It is important therefore that the protocol parameters that is used contain the sorts of value that reflect the current state of the ledger.
+
+Unfortunately, the upcoming Conway era introduces many new protocol parameters and obsoletes many others, meaning that there is no way to construct sensible values for these protocol parameters that would guarantee that transactions derived from them operate in a predictable way on chain.
+
+This would mean that submitting transactions built for an older era inherently carries with the risk that the transaction behaves other than intended.
+
+This is especially true for transactions built in an older era with an incomplete, made up or out-of-date values.
+
+This is also true for transactions upgraded from an older era, although to a lesser extent.
+
+Constructing a command in the currently era is the for any command that requires protocol-parametes is the only way to guarantee a transaction works as intended.
+
+Another thing that has changed since [[ADR-1 Default eras for CLI commands]] was adopted is the intruction of an era-based command structure.
+
+The era-based command structure looks like this: `cardano-cli selected-era command sub-command ...`.  For example `cardano-cli babbage transaction build ...`.
+
+We will for a time continue to support commands of the structure `cardano-cli command sub-command --selected-era` up until the Babbage era, but for Conway onwards, the era-based command structure must be used.
+
+The era-based command structure means that users when running a command must make some kind of decision upfront about what era they wish to run their command in.
+
+Such a structure also affords the possibility of having era selectors that follow an upgrade policy.  For example the `latest` era may select latest era that the CLI supports as a hard-coded value, defaulting to `babbage` whilst `conway` support is still in development.
+
+This ADR concerns what era selectors we will have and how they will behave.
+
+The decision should as much as possible take into account the following considerations:
+
+* Some users may care that CLI behaviour not suddenly change after a hard fork in a way that can break third party tools and user expectations.
+* Some users may care that CLI behaviour be such that transactions always behave as expected on-chain at the cost of breaking at a hard fork.
+* Historical precedent so as to avoid surprising SPOs and in a new release.
+* Simplicity and consistency.
+* Behaviour that just works where it makes sense.
+
+# Decision
+
+There will be no "default" era.  Instead, two era selectors will be included with the CLI:
```

Comment:
Does it mean that there will be only three eras available to select? What If I start a testnet in babbage, and I'd like to execute babbage commands. This new era selector will not allow for that.